ldap/servers/slapd/back-ldbm/idl_new.c

```diff
@@ -670,9 +703,294 @@ idl_new_range_fetch(
         }
         slapi_ch_free((void **)&leftover);
     }
+    slapi_log_err(SLAPI_LOG_FILTER, "idl_new_range_fetch",
+                  "Found %d candidates; error code is: %d\n",
+                  idl ? idl->b_nids : 0, *flag_err);
     return idl;
 }
 
+/*
+ * Callback used by idl_lmdb_range_fetch to add a new id in the id list
+ */
+static int
+idl_range_add_id_cb(dbi_val_t *key, dbi_val_t *data, void *ctx)
+{
+    idl_range_ctx_t *rctx = ctx;
+    int idl_rc = 0;
+    ID id = 0;
+
+    if (key->data == NULL) {
+        slapi_log_err(SLAPI_LOG_TRACE, "idl_range_add_id",
+                      "Unexpected empty key while iterating on %s index cursor\n", rctx->index_id);
+        return DBI_RC_NOTFOUND;
+    }
+    /* Stop iterating when reaching the upperkey */
+    if ((rctx->upperkey != NULL) && (rctx->upperkey->data != NULL)) {
+        if ((rctx->operator & SLAPI_OP_RANGE) == SLAPI_OP_LESS) {
+            if (keycmp(key, rctx->upperkey, rctx->ai->ai_key_cmp_fn) >= 0) {
+                return DBI_RC_NOTFOUND;
+            }
+        } else {
+            if (keycmp(key, rctx->upperkey, rctx->ai->ai_key_cmp_fn) > 0) {
+                return DBI_RC_NOTFOUND;
+            }
+        }
+    }
+    /* Check limits */
+    if ((rctx->lookthrough_limit != -1) &&
+        (rctx->idl->b_nids > (ID)rctx->lookthrough_limit)) {
+        idl_free(&rctx->idl);
+        rctx->idl = idl_allids(rctx->be);
+        slapi_log_err(SLAPI_LOG_TRACE, "idl_range_add_id", "lookthrough_limit exceeded\n");
+        rctx->flag_err = LDAP_ADMINLIMIT_EXCEEDED;
+        return DBI_RC_NOTFOUND;
+    }
+    if ((rctx->sizelimit > 0) && (rctx->idl->b_nids > (ID)rctx->sizelimit)) {
+        slapi_log_err(SLAPI_LOG_TRACE, "idl_range_add_id", "sizelimit exceeded\n");
+        rctx->flag_err = LDAP_SIZELIMIT_EXCEEDED;
+        return DBI_RC_NOTFOUND;
+    }
+    if ((rctx->idl->b_nids & 0xff) == 0 &&
```

Comment:
Do you mean SLAPI_OP_RANGE here?

Reply:
No, only testing the time limit every 256 candidates to speed up things